nxcomp/Log.h

```diff
@@ -198,18 +286,68 @@ NXLog& operator<< (NXLog& out, const NXLogStamp& value);
 
 
 template <typename T>
+bool has_newline(T value)
+{
+	return false;
+}
+
+template <char*>
+static bool has_newline(char *value)
+{
+	return strstr(value, "\n") != NULL;
```

Comment:
One less specific implementation, I'd say, I guess it should be a maintenance gain.